File: hackerrank/diagonalDifference.js
function diagonalDifference(arr) {
  let diagonal1 = 0;
  let diagonal2 = 0;
  //2D matrix. loop through and find the elements where i and j loop meet for diagonal1. Find the elements where the sum of i and j matches n-1. These indices are always the one you are looking for. i.e [0][2],[1,2], [2,1], [3,0] all sum up to three and would be the desired indices.
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr.length; j++) {
      if (i === j) diagonal1 += arr[i][j];
      if (i + j === arr.length - 1) diagonal2 += arr[i][j];
    }
  }
  return Math.abs(diagonal1 - diagonal2);
}

/* 
Given a square matrix, calculate the absolute difference between the sums of its diagonals. 
*/

File: python/hackerrank_1_week_prep/diagonal_difference.py
# Given a square matrix, calculate the absolute difference between the sums of its diagonals.

# For example, the square matrix is shown below:

# 1 2 3
# 4 5 6
# 9 8 9  
# Complete the 'diagonalDifference' function below.
#
# The function is expected to return an INTEGER.
# The function accepts 2D_INTEGER_ARRAY arr as parameter.
#
def diagonalDifference(arr):
    # Write your code here
    left_sum = 0
    right_sum = 0
    n = len(arr)
    for i in range(n):
        for j in range(n):
            if i == j:
                left_sum+= arr[i][j]
            if i+j == n-1:
                right_sum+= arr[i][j]
    return abs(left_sum - right_sum)